src/Arcus.Messaging.Abstractions.ServiceBus/MessageHandling/AzureServiceBusMessageRouter.cs

```diff
@@ -261,7 +248,12 @@ protected AzureServiceBusMessageRouter(IServiceProvider serviceProvider, AzureSe
                         var args = new ProcessMessageEventArgs(message, messageReceiver, cancellationToken);
                         SetServiceBusPropertiesForSpecificOperations(messageHandler, args, messageContext);
 
-                        await messageHandler.ProcessMessageAsync(result.DeserializedMessage, messageContext, correlationInfo, cancellationToken);
+                        bool isProcessed = await messageHandler.ProcessMessageAsync(result.DeserializedMessage, messageContext, correlationInfo, cancellationToken);
+                        if (!isProcessed)
+                        {
+                            continue;
```

Comment:
Yeah, you're in a `foreach` loop, it's only not shown in the diff. 
This will make it more clear: https://github.com/arcus-azure/arcus.messaging/blob/316f4cc9dca66a2e46d6b728b30b1ecb5328fcf8/src/Arcus.Messaging.Abstractions.ServiceBus/MessageHandling/AzureServiceBusMessageRouter.cs#L254